### PR TITLE
Signup works

### PR DESCRIPTION
**Pull request by isnotafunction**
relates #45 and #47
@Jbarget, thank you!

### CODE REVIEW

File: src/controllers/signup.js [hunk 1]
```diff
@@ -7,12 +7,18 @@ exports.get = (req, res) => {
...
 exports.post = (req, res) => {
   const { username, password } = req.body;
-  newUser.create({ username, password })
-    .then(console.log('Success!'))
-    .catch((error) => {
-      console.log(error);
+  bcrypt
+    .hash(password, 10)
+    .then(password => newUser.create({ username, password }))
+    .then(() => res.redirect('/thankYou'))
```
Comment (Jbarget) [inline src/controllers/signup.js]: might be nice to render `thankYou` with the user's username? just an idea no worries if not

`<h1>Thanks for signing up {{username}}!</h1> `

File: src/controllers/signup.js [hunk 2]
```diff
@@ -7,12 +7,18 @@ exports.get = (req, res) => {
...
+    .catch((err) => {
+      if (err.message.includes('duplicate')) {
+        console.log('this is err', err);
+        res.render('signup', { err: 'This username is already taken!' });
+      } else {
+        console.log(err);
```
Comment (Jbarget) [inline src/controllers/signup.js]: ideally we show the user an error page or message here

File: src/controllers/signup.js [hunk 3]
```diff
@@ -7,12 +7,18 @@ exports.get = (req, res) => {
...
-    .then(console.log('Success!'))
-    .catch((error) => {
-      console.log(error);
+  bcrypt
+    .hash(password, 10)
+    .then(password => newUser.create({ username, password }))
+    .then(() => res.redirect('/thankYou'))
+    .catch((err) => {
+      if (err.message.includes('duplicate')) {
+        console.log('this is err', err);
```
Comment (Jbarget) [inline src/controllers/signup.js]: remove this `console.log` please 🔥

File: public/signupValidation.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+const checkUsername = () => {
+  if (username.validity.valueMissing) {
+    displayErr(usernameErr, 'Please enter a username');
+  } else {
+    displayErr(usernameErr, '');
+    return true;
+  }
+};
+
+const checkPw = () => {
```
Comment (Parissai) [inline public/signupValidation.js]: Here we also need to check if the pw is less than 8 char

**Pull request merged**